### DISCUSSION TITLE
Getting more values than specified in predictions

### DISCUSSION THREAD

**Discussion by pranavlal** [amazon-science/chronos-forecasting#59] [Q&A] [answered]
Hi all,

I have specified a prediction length of 22 for my time series  but I am getting over a thousand values. I suspect I am using the model incorrectly. How do I correct my mistake? Here is the relevant portion of my code.

I am following the tutorial at the below link.
[](https://towardsdatascience.com/how-to-effectively-forecast-time-series-with-amazons-new-time-series-forecasting-model-9e04d4ccf67e
)

```

def load_model():
	pipeline = ChronosPipeline.from_pretrained("amazon/chronos-t5-base",device_map="cuda",torch_dtype=torch.bfloat16,)
	return pipeline

def predict(pipeline, timeseries, prediction_length=12):
	"""given a timeseries, predict with chronos model"""
	forecast = pipeline.predict(timeseries, prediction_length, num_samples=1)[0][0]  # shape [num_series, num_samples, prediction_length]
	return forecast

ts = torch.tensor(df["returns"].to_numpy(), dtype=torch.float32)
	pipeline = load_model()
	forecast = predict(pipeline, ts,64)
	npa=ts.detach().tolist()
	print(len(npa))
	
```

**Comment by lostella** [comment, marked as answer]
Hi @pranavlal, it looks like here you're printing the size of the input series, not of the forecast

```python
npa=ts.detach().tolist()
print(len(npa))
```

**Comment by pranavlal** [comment]
Hi @ Lorenzo,

 

Thanks, I wanted to count the number of entries in the forecast to determine if I was getting the expected number of entries.

**Reply by lostella** [reply]
For that you should do

```python
print(forecast.shape)
```

in your code above
